aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -161,10 +161,76 @@ static void where_kernel_impl(TensorIterator &iter, ScalarType condition_type) {
   });
 }
 
+static void isposinf_kernel_for_bool_impl(TensorIterator& iter) {
+  if (c10::isIntegralType(iter.input_dtype(), /*include_bool=*/true)) {
+    AT_DISPATCH_INTEGRAL_TYPES_AND(at::ScalarType::Bool, iter.input_dtype(), "isposinf_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a) -> bool { return false; });
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.input_dtype(), "isposinf_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a) -> bool { return a == std::numeric_limits<scalar_t>::infinity(); });
+    });
+  }
+}
+
+static void isposinf_kernel_for_scalar_impl(TensorIterator& iter) {
+  if (c10::isIntegralType(iter.input_dtype(), /*include_bool=*/true)) {
+    AT_DISPATCH_INTEGRAL_TYPES_AND(at::ScalarType::Bool, iter.input_dtype(), "isposinf_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a) -> scalar_t { return false; });
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.input_dtype(), "isposinf_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a) -> scalar_t { return a == std::numeric_limits<scalar_t>::infinity(); });
+    });
+  }
+}
+
+static void isposinf_kernel_impl(TensorIterator& iter) {
```

Comment:
I think you'll be able to get rid of this helper if the dispatch is simplified using the isfinite trick

Reply:
addressed, no more helpers are needed.